core/src/main/java/com/digitalpebble/storm/crawler/filtering/basic/BasicURLNormalizer.java

```diff
@@ -105,35 +107,45 @@ public void configure(Map stormConf, JsonNode paramNode) {
      * content.
      */
     private String filterQueryElements(String urlToFilter) {
-        URIBuilder uriBuilder;
         try {
             // Handle illegal characters by making a url first
             // this will clean illegal characters like |
             URL url = new URL(urlToFilter);
-            // this constructor will encode illegal characters.
-            URI uri =
-                    new URI(url.getProtocol(), url.getAuthority(), url.getPath(), url.getQuery(),
-                            url.getRef());
-            uriBuilder = new URIBuilder(uri);
-        } catch (URISyntaxException | MalformedURLException e) {
-            LOG.warn("Invalid urlToFilter {}. {}", urlToFilter, e);
-            return null;
-        }
 
-        List<NameValuePair> params = uriBuilder.getQueryParams();
-        List<NameValuePair> paramsToKeep = new ArrayList<>();
-        for (NameValuePair param : params) {
-            if (!queryElementsToRemove.contains(param.getName())) {
-                paramsToKeep.add(param);
+            if (StringUtils.isEmpty(url.getQuery())) {
+                return urlToFilter;
+            }
+
+            List<NameValuePair> pairs = new ArrayList<NameValuePair>();
+            URLEncodedUtils.parse(pairs, new Scanner(url.getQuery()), "UTF-8");
```

Comment:
Several answers for you:
- In absolute no, in practice yes
- the same % that was assumed before my code change since UTF-8 is assumed when you construct a URI from a URL since URI are in UTF-8 always but URL don't have to be.